### PR TITLE
mysqldef: save FSP and fix zone offset for codec

### PR DESCRIPTION
**Pull request by siddontang**
@coocood  @qiuyesuifeng

### CODE REVIEW

File: mysqldef/time.go [hunk 1]
```diff
@@ -175,11 +199,17 @@ func (t *Time) UnmarshalInLocation(b []byte, loc *time.Location) error {
...
 	if t.Type == TypeDatetime || t.Type == TypeDate {
 		// e.g, for 2010-10-10T10:10:10 UTC, we will unmarshal to 2010-10-10T10:10:10 location
-		_, offset := time.Now().In(loc).Zone()
+		// We must use Date to creates a time to get offset. Why not time.Now directly,
+		// because creating time using Date with time.Local may still have a different zone with time.Now.
+		year, month, day := t.Time.Date()
+		hour, min, sec := t.Time.Clock()
+		nsec := t.Time.Nanosecond()
+		_, offset := time.Date(year, month, day, hour, min, sec, nsec, time.Local).Zone()
+
```
Comment (coocood) [inline mysqldef/time.go]: How about `_, offset := t.Time.In(loc).Zone()`?

Comment (siddontang) [inline mysqldef/time.go]: yes, it works

**Comment by coocood** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**